xrpl/transaction/types/credential_type.go

```diff
@@ -21,16 +33,75 @@ func (c *CredentialType) String() string {
 // - Not empty
 // - Valid hex string
 // - Length between MinCredentialTypeLength and MaxCredentialTypeLength
-func (c *CredentialType) IsValid() bool {
-	if c.String() == "" {
+func (c CredentialType) IsValid() bool {
+	hexStr := c.String()
+	if hexStr == "" {
 		return false
 	}
-
-	credTypeStr := c.String()
-	if !typecheck.IsHex(credTypeStr) {
+	if !typecheck.IsHex(hexStr) {
 		return false
 	}
-
-	length := len(credTypeStr)
+	length := len(hexStr)
 	return length >= MinCredentialTypeLength && length <= MaxCredentialTypeLength
 }
+
+// AuthorizeCredential represents an accepted credential for PermissionedDomainSet transactions.
+type AuthorizeCredential struct {
+	Issuer         Address
```

Comment:
`AuthorizedCredential` contains a field `Credential` which contains this ones. You can take a look to xrpl.js to check it.